sdk/core/azure_core_amqp/Cargo.toml

```diff
@@ -29,6 +29,9 @@ serde.workspace = true
 serde_amqp = { workspace = true, optional = true }
 serde_bytes = { workspace = true, optional = true }
 tokio.workspace = true
+tokio-socks = "0.5"
+tokio-native-tls = "0.3"
+native-tls = "0.2"
```

Comment:
These should be defined as part of an optional feature for anyone to enable who wants to use the protocol. Most customers likely won't need it and the increased build time required.

@LarryOsterman we'd also want to consider whether these should be declared directly herein or in the workspace `Cargo.toml`. The only net new dependency is `tokio-socks`, but I suspect that - at least for now - AMQP is the only protocol that would support it. The others are already in the `Cargo.lock` file and should just be in the workspace if they even need to be declared explicitly. `fe2o3-amqp@0.14.0` already has a dependency on `native-tls@0.2.14` and `tokio-native-tls@0.3.1` so they probably don't need to be in here at all.